packages/next/types/index.d.ts

```diff
@@ -65,17 +65,30 @@ export {
   NextApiHandler,
 }
 
-export type GetStaticProps<
-  P extends { [key: string]: any } = { [key: string]: any },
-  Q extends ParsedUrlQuery = ParsedUrlQuery
-> = (ctx: {
+export type GetStaticPropsContext<Q extends ParsedUrlQuery = ParsedUrlQuery> = {
   params?: Q
   preview?: boolean
   previewData?: any
-}) => Promise<{
+}
+
+export type GetStaticPropsReturnType<P> = {
```

Comment:
It feels weird when a type has `type` in its name, can we rename this one? maybe `GetStaticPropsResult` or something similar :thinking: